File: split-with-minimum-sum-2578/solution.js
/*
 * LeetCode problem # 2578, 'Split With Minimum Sum'
 * https://leetcode.com/problems/split-with-minimum-sum
 */

/**
 * @param {number} num
 * @return {number}
 */
var splitNum = function (num) {
  const s = (num + "").split("");
  s.sort();

  const a = s.filter((c, i) => i % 2 === 0).join("");
  const b = s.filter((c, i) => i % 2 !== 0).join("");

  return parseInt(a) + parseInt(b);
};

File: split-with-minimum-sum-2578/Solution.java

/**
 * LeetCode problem # 2578, 'Split With Minimum Sum'
 * https://leetcode.com/problems/split-with-minimum-sum
 */

import java.util.Arrays;

class Solution {
    public int splitNum(int num) {
        char[] chars = String.valueOf(num).toCharArray();
        Arrays.sort(chars);

        StringBuilder a = new StringBuilder();
        StringBuilder b = new StringBuilder();

        for (int i = 0; i < chars.length; ++i) {
            if (i % 2 == 0) {
                a.append(chars[i]);
            } else {
                b.append(chars[i]);
            }
        }
        return Integer.parseInt(a.toString()) + Integer.parseInt(b.toString());
    }
}
